### PR TITLE
Use vscode.workspace.fs instead of fs package

### LINKED ISSUE #812: Use VS Code `workspace.fs` for filesystem operations
Follow up to #739 

--- 

## Problem Statement

Currently the extension uses the NodeJS "native" `fs` module to perform filesystem operations.

VS Code API provides [workspace.fs](https://code.visualstudio.com/api/references/vscode-api#FileSystem) for reading and writing files and folders from an uri.

It is possible that using the native VS Code FS API _could_ alleviate some of the FS related issues users have reported, such as https://github.com/hashicorp/vscode-terraform/issues/732

## Expected User Experience

No UX changes expected.

## Proposal

TODO

### PR DESCRIPTION
**Pull request by jpogran**
Replaces all usages of the fs package with the vscode.workspace.fs wrapper. This ensures safe usage of the filesystem and allows us in the future to move to remote extension use.

Fix bug where zip is not deleted after extraction. This also helps in the future when we minify the extension and del does not work.

Closes #812

### CODE REVIEW

File: src/languageServerInstaller.ts [hunk 1]
```diff
@@ -70,7 +68,7 @@ export class LanguageServerInstaller {
       // Most of the time, getLsVersion will produce "ENOENT: no such file or directory"
```
Comment (dbanck) [inline src/languageServerInstaller.ts:68]:
```suggestion
      // Most of the time, getLsVersion will produce "FileSystemError: FileNotFound"
```
I think `workspace.fs` is throwing `FileSystemError`s: https://code.visualstudio.com/api/references/vscode-api#FileSystemError

**Review by dbanck — APPROVED** [review]
Installation works well for me, and the `.zip` file is deleted, too 🎉 

I've two minor suggestions but feel free to merge anyway.

**Comment by radeksimko** [comment]
> Replaces all usages of the fs package with the vscode.workspace.fs wrapper.

Just for posterity, some packages that the extension depends on still use the `fs` package, such as `js-releases`.
I'm assuming that could be addressed of as part of future work on bundling of LS though.

👍🏻 🚢

**Comment by jpogran** [comment]
Updated commit message 😝 

> Replaces all usages _*inside extension*_ of the fs package with the `vscode.workspace.fs` wrapper.

**Pull request merged**